conda/base/context.py

```diff
@@ -871,7 +874,11 @@ def _get_user_agent(context_platform):
     libc_family, libc_ver = linux_get_libc_version()
     if context_platform == 'linux':
         from .._vendor.distro import linux_distribution
-        distinfo = linux_distribution(full_distribution_name=False)
+        try:
+            distinfo = linux_distribution(full_distribution_name=False)
+        except Exception as e:
+            log.debug('%r', e, exc_info=True)
+            distinfo = ('Linux', 'unknown')
```

Comment:
This part has nothing to do with the PR.  Just got snuck in.  The `linux_distribution` function from the vendored `distro` can fail (it also makes a subprocess call to `lsb_release`, which I didn't know).  Not all that happy with how it works.  But oh well.

Reply:
Ok!

For some reason I usually prefer to do things like

```python
distinfo = ('Linux', 'unknown')
try:
    distinfo = linux_distribution(full_distribution_name=False)
except Exception as e:
    log.debug('%r', e, exc_info=True)
```

I know they have the same meaning :-p

Reply:
On this style issue, can we agree to disagree?

I feel like the way it is here is more idiomatic for try/except. Try, and then on failure, here’s how to recover. Rather than the default value case.